### PR TITLE
Hw fifth part spring boot data validation

### PR DESCRIPTION
**Pull request by pikador10**

### CODE REVIEW

File: src/main/java/com/hw/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    public BookDto createBook(@RequestBody @Valid BookRequestDto requestDto) {
+        return bookService.save(requestDto);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable Long id,
+                                  @RequestBody BookRequestDto requestDto) {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/controller/BookController.java]: Valid annotation missed

File: src/main/java/com/hw/bookstore/dto/BookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import jakarta.validation.constraints.DecimalMax;
+import jakarta.validation.constraints.DecimalMin;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import java.math.BigDecimal;
+
+public record BookRequestDto(
+
+        @NotNull(message = "Title can not be null")
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/dto/BookRequestDto.java:11]: Better use NotBlank with literals. Here you can find info about the difference https://www.baeldung.com/java-bean-validation-not-null-empty-blank

File: src/main/java/com/hw/bookstore/dto/BookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        @Size(min = 1, max = 255, message = "Title size must be between 1 and 255 symbols")
+        String title,
+
+        @NotNull(message = "Author can not be null")
+        @Size(min = 1, max = 255, message = "Author size must be between 1 and 255 symbols")
+        String author,
+
+        @NotNull(message = "Isbn can not be null")
+        @Size(min = 1, max = 255, message = "Isbn size must be between 1 and 255 symbols")
+        String isbn,
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/dto/BookRequestDto.java:21]: You can add specific Isbn validation (use pattern or create your own annotation)

File: src/main/java/com/hw/bookstore/exception/handler/ErrorBody.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ErrorBody {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/exception/handler/ErrorBody.java:14]: Why Error? Let's use Exception ;)

File: src/main/java/com/hw/bookstore/exception/handler/ErrorMessage.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ErrorMessage {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/hw/bookstore/exception/handler/ErrorMessage.java:12]: the same

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! A few minor comments;)

File: src/main/java/com/hw/bookstore/controller/BookController.java (hunk 1 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/controller/BookController.java]: My mistake. Annotation has been added.

File: src/main/java/com/hw/bookstore/dto/BookRequestDto.java (hunk 2 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/dto/BookRequestDto.java:11]: Yes, it's a great comment. But I've added size validation and it already checks for empty, so we just need to check for null.

File: src/main/java/com/hw/bookstore/dto/BookRequestDto.java (hunk 3 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/dto/BookRequestDto.java:21]: That's great advice, I'll think about it.

File: src/main/java/com/hw/bookstore/exception/handler/ErrorBody.java (hunk 4 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/exception/handler/ErrorBody.java:14]: Ohh, I use my exception response pattern, so I can manage my response format using ErrorBody. Let's keep this format

File: src/main/java/com/hw/bookstore/exception/handler/ErrorMessage.java (hunk 5 above)
Comment (pikador10) [reply src/main/java/com/hw/bookstore/exception/handler/ErrorMessage.java:12]: Let's keep this format ))

File: src/main/java/com/hw/bookstore/dto/BookRequestDto.java (hunk 2 above)
Comment (liliia-ponomarenko) [reply src/main/java/com/hw/bookstore/dto/BookRequestDto.java:11]: annotation size will not validate a value like "     "

**Review by liliia-ponomarenko — APPROVED** [review]
Great job! 😉

**Pull request merged**